lib/index.js

```diff
@@ -354,12 +397,25 @@ exports.deleteFunction = async (ctx, boundaryId, functionId) => {
             boundaryId || ctx.body.boundaryId
         }/function/${functionId || ctx.body.functionId}`
     ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+
+    // Always attempt to clean up storage
+    await exports.deleteStorage(ctx);
 };
 
-exports.getStorageClient = (ctx) => {
+exports.getFunctionUrl = async (ctx, boundaryId, functionId) => {
+    let response = await Superagent.get(
+        `${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/subscription/${ctx.body.subscriptionId}/boundary/${
+            boundaryId || ctx.body.boundaryId
+        }/function/${functionId || ctx.body.functionId}/location`
+    ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+
+    return response.body.location;
+}
+
+getStorageClient = (ctx) => {
     const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
         algorithm: 'RS256',
-        expiresIn: 60 * 60 * 24,
+        expiresIn: 60 * 15, // 15 min makes more sense given Lambda lifecycle
```

Comment:
Maybe add 30s/1m to mitigate any race conditions towards the end of the life of the Lambda (the token is created before the lambda starts up)